main.cpp

```diff
@@ -8762,6 +8772,13 @@ int main(int argc, char **argv) {
                         // again is to assume it has the same serial number.
                         settings.address = -1;
                         settings.bus = -1;
+                        if (settings.pid != -1 || settings.vid != -1) {
+                            // also skip vid/pid filtering, as that should change in BOOTSEL mode, and could be white-labelled on RP2350
+                            settings.pid = -1;
+                            // still filter for rpi vid/pid if we don't have a serial number, as that is an RP2040 running a no_flash binary, so will
+                            // have a standard rpi vid/pid in BOOTSEL mode
+                            settings.vid = settings.ser.empty() ? -1 : 0;   // 0 means skip vid/pid filtering entirely, -1 means filter for rpi vid/pid
```

Comment:
Perhaps something like this is slightly less confusing? :man_shrugging: 
```suggestion
                            if (settings.ser.empty()) {
                                // still filter for rpi vid/pid if we don't have a serial number, as that is an RP2040 running a no_flash binary, so will
                                // have a standard rpi vid/pid in BOOTSEL mode
                                settings.vid = -1; // -1 means filter for rpi vid/pid
                                settings.pid = -1;
                            } else {
                                // also skip vid/pid filtering, as that should change in BOOTSEL mode, and could be white-labelled on RP2350
                                settings.vid = 0; // 0 means skip vid/pid filtering entirely
                                settings.pid = -1;
                            }
```

Reply:
Have resolved this with something similar to this